src/gwas/src/gwas/plots/helpers.py

```diff
@@ -0,0 +1,118 @@
+# -*- coding: utf-8 -*-
+import pickle
+from pathlib import Path
+from typing import List, Tuple
+
+import matplotlib.pyplot as plt
+import pandas as pd
+import zstandard as zstd
+from qmplot import manhattanplot
+from scipy.stats.distributions import chi2
+
+
+def chi2_pvalue(ustat: int | float, vstat: int | float) -> float:
+    """Calculates the p-value from U-statistic and V-statistic using the chi-square test."""
+    chi2_stat = (ustat**2) / vstat
+    p_value = chi2.sf(chi2_stat, df=1)
+    return p_value
+
+
+def find_phenotype_index(phenotype_label: str, phenotypes_list: List) -> Tuple[int, int]:
+    """
+    Finds the indices of the U-statistic and V-statistic for a given phenotype label.
+    Args:
+        phenotype_label (str): The label of the phenotype to find indices for.
+        phenotypes_list (List[str]): A list of phenotype descriptors.
+
+    Returns:
+        Tuple[int, int]: A tuple containing the indices of the U-statistic and V-statistic.
+
+    Raises:
+        ValueError: If either the U-statistic or V-statistic index cannot be found for the
+        specified phenotype label.
+    """
+    u_stat_idx = None
+    v_stat_idx = None
+    # namedTuple für phenotype
+    for index, phenotype in enumerate(phenotypes_list):
+        if phenotype_label in phenotype:
+            if "stat-u" in phenotype:
+                u_stat_idx = index
+            elif "stat-v" in phenotype:
+                v_stat_idx = index
+
+    if u_stat_idx is not None and v_stat_idx is not None:
+        return u_stat_idx, v_stat_idx
+    else:
+        raise ValueError(
+            f"Matching indices for '{phenotype_label}' not found. u_stat_idx: {u_stat_idx}, v_stat_idx: {v_stat_idx}"
+        )
+
+
+def load_metadata(metadata_path: Path):
+    with open(metadata_path, "rb") as f:
```

Comment:
I would prefer if we could use the `gwas.compression.pipe.CompressedBytesReader` here to simplify the code, see for example
https://github.com/HippocampusGirl/GWASProtocol/blob/226aeb5f09bbe71abe7e7061a64363f26f915a96/src/gwas/tests/compression/test_pipe.py#L33-L34

Reply:
Updated:

```
def load_metadata(metadata_path: Path):
    with CompressedBytesReader(metadata_path) as f:
```